.github/ISSUE_TEMPLATE/spec_request.md

```diff
@@ -0,0 +1,55 @@
+---
+title: 'Request creation of a new format specification'
+labels: spec-request
+---
+<!--
+A request for a format specification consists of 2 YAML blocks followed by a free-form text description.
+Fill in all the fields possible. Delete fields you cannot populate and all the comments from this template (including this one!).
+The resulting YAML blocks should contain valid YAML.
+Mandatory fields are marked with the word MANDATORY for your convenience.
+-->
```

Comment:
This whole template is way too long, all this bureaucracy will be a major turn off for potential people who will want to create an issue. I'll post some suggestions with inline comments.

Reply:
>This whole template is way too long, all this bureaucracy will be a major turn off for potential people who will want to create an issue.

Any restriction is a turn-off for people wanting to create an issue. But the issues like "i wanna owners to create me a spec for a format I don't remember name of but I know it is used in 3d graphics and has over 9k users" are useless junk.

IMHO there are 2 use cases for `spec-request` issues:
* a person is working on the format himself and already has all the needed info. Then he already has all the needed info;
* a person is not working on the format, but he may want in future, if noone has implemented it untill then, so he dumps the info for himself.

IMHO everyone else is not intended to create `spec-request` issues because unimplemented ideas worth nothing. If someone needs a format, he will implement it. Otherwise it will never be implemented. And if a person is not ready to spend even a bit of his time (few seconds!) to find the sources and fill in the form, but is ready only to ask, it means he doesn't need that format at all and it is very unlikely that he will implement it (usually takes some hours which are orders of magnitude more than seconds). So this will result into a junk issue (an issue without any useful info about the format really needed by noone, even the one who had created it didn't need it) which IMHO should be automatically removed from the repo. So when a person who needs  it enough to spend at least a few seconds will come, he will create the properly formatted issue, not the junk one.

Reply:
> "i wanna owners to create me a spec for a format I don't remember name of but I know it is used in 3d graphics and has over 9k users"

This almost never happens in practice. The majority of people who bother to report an issue really do care about it and will do the best they can to provide all relevant info. A template is great for this purpose, because it tells/reminds people about what should be included in the issue.

There will inevitably be low-effort issues at some point, but they are really rare (when is the last time you saw a truly useless issue on the main kaitai_struct repo?), and those low-effort posters will probably ignore the template anyway, no matter how many comments and messages you add.

Reply:
>those low-effort posters will probably ignore the template anyway, no matter how many comments and messages you add.

The point of having the template is to automatically enforce it. So the issues with this label not matching the template should be automatically removed at least from this repo if not fixed after some time.

Reply:
> because unimplemented ideas worth nothing

I strongly disagree with this. I think it's totally valid to create a "new format" issue even if you have no intention of implementing it yourself. Obviously the issue needs to contain the necessary relevant info, there's no question about that. But I don't think that making a "new format" issue should require a commitment from the reporter that they will implement the format at some point.

Reply:
It doesn't require a commitment. But if even the reporter didn't bother to spend a few seconds, it seems the format is really not needed even by its reporter - the person who is supposed to need it the most of all other people in the repo: If it had ever been needed by someone in the repo more than the issue reporter, the issue would have already been already there. So if noone needs it, it will be never implemented.

Of course one can say "what if someone found that issue in Google searching for the format, tot known about Kaitzi and implemented it himself"? A completely valid concern, but ... I don't think that Google will range the unfilled template high enough.

Reply:
> The point of having the template is to automatically enforce it.

Um, no. The point of the template is to get people to include all important info in a "new format" issue. "Enforcing" the template by automatically deleting misformatted issues is missing the point. An automated format checker for these issues might be useful in the future, but it's not required, and it's definitely not the *reason* why we're adding the template.

> So the issues with this label not matching the template should be automatically removed at least from this repo if not fixed after some time.

As has already been mentioned in kaitai-io/kaitai_struct#134, deleting misformatted issues is an unnecessarily strong reaction. Just tag the issue as "not formatted properly" or whatever and *maybe* close it if it isn't fixed after a while, but definitely don't *delete* the issue because of something so trivial as a formatting error.

> But if even the reporter didn't bother to spend a few seconds, it seems the format is really not needed even by its reporter

Yes, we have already established that, and I completely agree - low-effort issues with no useful info should be closed. That is not what I am talking about. I am talking about issues that *do* contain useful information and are properly formatted, but which (for whatever reason) the issue reporter won't implement themselves (but somebody else might). Do you think those issues should be allowed, and if not, why?

Reply:
>Um, no. The point of the template is to get people to include all important info in a "new format" issue.

Both yes and no. As you have said, people unwilling to instert the info can just delete the template, or even worse, leave it as it is (and this happens often in some repos - someone just comes and creates an issue consisting of an unfilled template entirely). An adequate template per-se just encourages people to fill it in. It is ineffective when people are unwiling to fill it in. What effectively makes them to fill it in (in fact it doesnt, but the result is the same) is enforcing the template by cleaning somehow the mess they have created.

>Do you think those issues should be allowed, and if not, why?

They should be allowed since they are [prooven to be needed (since the poster had invested some labour into them)](https://en.wikipedia.org/wiki/Proof-of-work) (of course it is theoretically possible to hack this bot once by generating junk issues automatically and sending them from mass-registered accounts, but IMHO noone gonna waste own time on that) and contain useful info.

>As has already been mentioned in kaitai-io/kaitai_struct#134, deleting misformatted issues is an unnecessarily strong reaction. Just tag the issue as "not formatted properly" or whatever and maybe close it if it isn't fixed after a while, but definitely don't delete the issue because of something so trivial as a formatting error.

Yes, closing, moving, assigning a tag - anything that will remove the issue from the default search (and I would also like to remove it from my subscribtion feed untill the owner filled in all the needed info, but GH has no such a feature).